From d72e785c46305fe00eb126d315d2dfe4b98f458a Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Mon, 13 Oct 2014 11:50:26 +0200
Subject: [PATCH] Avoid duplicate MOJO parameters

---
 .../JavaAnnotationsMojoDescriptorExtractor.java    | 24 ++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/annotations/JavaAnnotationsMojoDescriptorExtractor.java b/maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/annotations/JavaAnnotationsMojoDescriptorExtractor.java
index d595438..ed3a17f 100644
--- a/maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/annotations/JavaAnnotationsMojoDescriptorExtractor.java
+++ b/maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/annotations/JavaAnnotationsMojoDescriptorExtractor.java
@@ -61,6 +61,7 @@ import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
@@ -554,7 +555,7 @@ public class JavaAnnotationsMojoDescriptorExtractor
                 parameter.setSince( parameterAnnotationContent.getSince() );
                 parameter.setRequired( parameterAnnotationContent.required() );
 
-                mojoDescriptor.addParameter( parameter );
+                addParameter( mojoDescriptor, parameter );
             }
 
             // Component annotations
@@ -594,7 +595,7 @@ public class JavaAnnotationsMojoDescriptorExtractor
                 //parameter.setRequired( ... );
                 parameter.setEditable( false );
 
-                mojoDescriptor.addParameter( parameter );
+                addParameter( mojoDescriptor, parameter );
             }
 
             mojoDescriptor.setPluginDescriptor( pluginDescriptor );
@@ -604,6 +605,25 @@ public class JavaAnnotationsMojoDescriptorExtractor
         return mojoDescriptors;
     }
 
+    private void addParameter( ExtendedMojoDescriptor mojoDescriptor,
+                               org.apache.maven.plugin.descriptor.Parameter parameter )
+        throws DuplicateParameterException
+    {
+        if ( mojoDescriptor.getParameters() != null )
+        {
+            for ( Iterator<?> it = mojoDescriptor.getParameters().iterator(); it.hasNext(); )
+            {
+                if ( it.next().equals( parameter ) )
+                {
+                    getLogger().warn( "Duplicate parameter " + parameter.getName() + " field in MOJO descriptor" );
+                    it.remove();
+                }
+            }
+        }
+
+        mojoDescriptor.addParameter( parameter );
+    }
+
     protected ExecuteAnnotationContent findExecuteInParentHierarchy( MojoAnnotatedClass mojoAnnotatedClass,
                                                                      Map<String, MojoAnnotatedClass> mojoAnnotatedClasses )
     {
-- 
1.9.3

