--- surefire-2.19/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java~	2015-10-14 21:15:22.000000000 +0200
+++ surefire-2.19/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java	2015-10-20 08:53:24.412715344 +0200
@@ -162,16 +162,7 @@
     private static String getFailureBehavior( MavenExecutionRequest request )
         throws NoSuchMethodException, InvocationTargetException, IllegalAccessException
     {
-        try
-        {
-            return request.getFailureBehavior();
-        }
-        catch ( NoSuchMethodError e )
-        {
-            return (String) request.getClass()
-                .getMethod( "getReactorFailureBehavior" )
-                .invoke( request );
-        }
+        return request.getReactorFailureBehavior();
     }
 
 }
