diff -up modello-core/src/main/java/org/codehaus/modello/Modello.java.sav modello-1.0-alpha-15/modello-core/src/main/java/org/codehaus/modello/Modello.java
--- modello-core/src/main/java/org/codehaus/modello/Modello.java.sav	2008-06-20 16:34:47.000000000 -0400
+++ modello-core/src/main/java/org/codehaus/modello/Modello.java	2008-06-20 16:36:02.000000000 -0400
@@ -30,8 +30,6 @@ import org.codehaus.modello.core.Modello
 import org.codehaus.modello.model.Model;
 import org.codehaus.modello.model.ModelValidationException;
 import org.codehaus.plexus.embed.Embedder;
-import org.codehaus.plexus.PlexusContainer;
-import org.codehaus.plexus.DefaultPlexusContainer;
 
 /**
  * @author <a href="mailto:jason@maven.org">Jason van Zyl</a>
@@ -40,7 +38,7 @@ import org.codehaus.plexus.DefaultPlexus
  */
 public class Modello
 {
-    PlexusContainer container;
+    Embedder embedder = new Embedder();
 
     ModelloCore core;
 
@@ -49,9 +47,9 @@ public class Modello
     {
         try
         {
-            container = new DefaultPlexusContainer();
+            embedder.start();
 
-            core = (ModelloCore) container.lookup( ModelloCore.ROLE );
+            core = (ModelloCore) embedder.lookup( ModelloCore.ROLE );
         }
         catch( Exception ex )
         {
