--- plexus-compilers/plexus-compiler-jikes/src/main/java/org/codehaus/plexus/compiler/jikes/JikesCompiler.java.sav	2006-05-30 13:06:25.000000000 +0200
+++ plexus-compilers/plexus-compiler-jikes/src/main/java/org/codehaus/plexus/compiler/jikes/JikesCompiler.java	2006-05-30 13:08:18.000000000 +0200
@@ -289,4 +289,9 @@
 
         return new CompilerError( file, type.equals( "error" ), startline, startcolumn, endline, endcolumn, message );
     }
+
+    public String[] createCommandLine(CompilerConfiguration config ) throws CompilerException
+    {
+        return null;
+    }
 }
